test/parallel/xyk-pallet.API.pool.deactivatedPool.test.ts

```diff
@@ -0,0 +1,316 @@
+    ).then((result) => {
+      const eventResponse = getEventResultFromMangataTx(result);
+      expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+    });
+
+    const deactivatedPoolBalance = await getBalanceOfPool(MGA_ASSET_ID, token1);
+
+    expect(deactivatedPoolBalance[0][0]).bnGt(BN_ZERO);
+  });
+
+  test("WHEN the user mints liquidity in the pool again THEN liquidity IDs are equal", async () => {
+    await mintLiquidity(
+      testUser2.keyRingPair,
+      MGA_ASSET_ID,
+      token1,
+      defaultCurrencyValue
+    ).then((result) => {
+      const eventResponse = getEventResultFromMangataTx(result);
+      expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+    });
+
+    const liquidityIdAfter = await getLiquidityAssetId(MGA_ASSET_ID, token1);
+
+    expect(liquidityIdAfter).bnEqual(liquidityId);
+  });
+
+  test("WHEN the user tries to swap/multiswap tokens on the deactivated pool THEN error returns", async () => {
+    let swapError: any = [];
+    try {
+      await sellAsset(
+        testUser1.keyRingPair,
+        MGA_ASSET_ID,
+        token1,
+        defaultCurrencyValue,
+        new BN(1)
+      );
+    } catch (error) {
+      swapError = error;
+    }
+    expect(swapError.data).toEqual(
+      "1010: Invalid Transaction: The swap prevalidation has failed"
+    );
+  });
+
+  test("WHEN the user tries to compound reward on a deactivated pool THEN error returns", async () => {
+    await compoundRewards(testUser1, liquidityId).then((result) => {
+      const eventResponse = getEventResultFromMangataTx(result);
+      expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicFailed);
+      expect(eventResponse.data).toEqual("NotAPromotedPool");
+    });
+  });
+
+  test("WHEN sudo try to promote a pool THEN poolPromotion is updated", async () => {
+    const api = getApi();
+
+    await promotePool(sudo.keyRingPair, liquidityId, 35).then((result) => {
+      const eventResponse = getEventResultFromMangataTx(result);
+      expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+    });
+
+    const poolRewards = JSON.parse(
+      JSON.stringify(await api.query.proofOfStake.promotedPoolRewards())
+    );
+    const poolWeight = stringToBN(poolRewards[liquidityId.toString()].weight);
+
+    expect(poolWeight).bnEqual(new BN(35));
+  });
+
+  test("WHEN a bootstrap is scheduled for the existing pair THEN the operation fail with pool already exist.", async () => {
+    await setupApi();
+
+    await checkLastBootstrapFinalized(sudo);
+    await setupBootstrapTokensBalance(token1, sudo, [testUser1]);
+
+    const sudoBootstrap = await scheduleBootstrap(
+      sudo,
+      MGA_ASSET_ID,
+      token1,
+      5,
+      5,
+      5
+    );
+    await waitSudoOperationFail(sudoBootstrap, ["PoolAlreadyExists"]);
+  });
+});
+
+describe("GIVEN user create a pool, wait for rewards and then deactivate the pool", () => {
+  beforeEach(async () => {
```

Comment:
generatign this setup is super time consuming. and only one test do real modifications. so perhaps you can leave this setup to `WHEN user tries to claim rewards THEN rewards are claimed` and add the validation of the rpc. for rewards . 
Others must go to the other describe, since we do not require rewards setup. 

The idea is to have only one test with the rewards setup. and the remaining tests can go to the previous describe.

Reply:
We also need to do it for `WHEN call RPC calculate_rewards_amount for this user THEN amount return`, I moved other tests

Reply:
perhaps we can calculate_rewards inside the Claiming rewards. Not the best, but we can save some time instead. By this we can even compare the rpc result with the total claimed.

Reply:
Done